python/servo/command_base.py

```diff
@@ -33,6 +35,54 @@ def cd(new_path):
         os.chdir(previous_path)
 
 
+@contextlib.contextmanager
+def setlocale(name):
+    """Context manager for changing the current locale"""
+    saved_locale = locale.setlocale(locale.LC_ALL)
+    try:
+        yield locale.setlocale(locale.LC_ALL, name)
+    finally:
+        locale.setlocale(locale.LC_ALL, saved_locale)
+
+
+def archive_deterministically(dir_to_package, dest_archive, prepend_path=None):
+    """Create a .tar.gz package in a deterministic (reproducible) manner.
+
+    See https://reproducible-builds.org/docs/archives/ for more details. """
+
+    def reset(tarinfo):
+        """Helper to reset owner/group and modification time for tar entries"""
+        tarinfo.uid = tarinfo.gid = 0
+        tarinfo.uname = tarinfo.gname = "root"
+        tarinfo.mtime = 0
+        return tarinfo
+
+    with cd(dir_to_package):
+        current_dir = "."
+        file_list = [current_dir]
+        for root, dirs, files in os.walk(current_dir):
+            for name in dirs + files:
+                file_list.append(os.path.join(root, name))
+
+        # Sort file entries with the fixed locale
+        with setlocale('C'):
+            file_list = sorted(file_list, cmp=locale.strcoll)
```

Comment:
Generally I like immutable values, but in this case I think using `file_list.sort()` instead of `sorted()` is an appropriate optimization since we create and consume `file_list` in this function and we won't use the original value ever again.